### ISSUE TITLE
Sending text does not work anymore after installing sendtext+

### ISSUE THREAD

**Issue by JanaJarecki** [randy3k/R-Box#75]
Hey there,
after installation of sendtext+, ctrl+enter and user-defined commands dont work anymore.
Any idea what causes that?
SublimeText Built 3103, Windows 10
Best
Jana

**Comment by randy3k** [comment]
Hi Jana,

Could you try to run `SendText+: choose program` in the command palette?

**Comment by JanaJarecki** [comment]
thanks  - so quick! :)

Result:
ctrl+enter works, sends, opens R
My user defined command to switch the working directory (see below) does 
not work

[
     {
         "keys": ["ctrl+alt+d"],
         "command": "r_box_change_dir",
         "context": [
             { "key": "selector", "operator": "equal", "operand": 
"source.r"}
         ]
     }
]


Am 24.02.2016 um 18:18 schrieb Randy Lai:
>
> Hi Jana,
>
> Could you try to run |SendText+: choose program| in the command palette?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/randy3k/R-Box/issues/75#issuecomment-188359209>.
>


-- 
Jana Jarecki (Researcher)
Economic Psychology Lab | University Basel
Missionsstr. 62A | CH-4055 Basel
janajaskype

**Comment by randy3k** [comment]
The send code function has out sourced to SendText+, the equivalent command is now
`send_text_plus_change_dir` instead of `r_box_change_dir`.

**Comment by JanaJarecki** [comment]
Works - marvellous. Thanks so much, as always!
Cheers,
Jana

Am 24.02.2016 um 18:25 schrieb Randy Lai:
>
> The send code function has out sourced to SendText+, the equivalent 
> command is now
> |send_text_plus_change_dir| instead of |r_box_change_dir|.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/randy3k/R-Box/issues/75#issuecomment-188361616>.
>


-- 
Jana Jarecki (Researcher)
Economic Psychology Lab | University Basel
Missionsstr. 62A | CH-4055 Basel
janajaskype

**Comment by randy3k** [comment]
Great

**Comment by fpegmbg** [comment]
Randy, Jana: I experienced the same send failure after SendText was installed, Win7, R3.2.2, SublimeText Build 3103. This posting solved the problem--I can't thank you enough.

**Comment by randy3k** [comment]
@JanaJarecki 

Please be noticed that in the near future. The source command would be changed to
```
    {
        "keys": ["ctrl+\\"], "command": "send_text_plus",
        "args": {"cmd": "setwd(\"$file\")"},
        "context": [{"key": "selector", "operator": "equal", "operand": "source.r"}]
    }
```
It allows more flexibility for users to define their own commands.

**Comment by randy3k** [comment]
@JanaJarecki 

Please be noticed that in the near future. The source command would be changed to
```
    {
        "keys": ["ctrl+\\"], "command": "send_text_plus",
        "args": {"cmd": "setwd(\"$file_path\")"},
        "context": [{"key": "selector", "operator": "equal", "operand": "source.r"}]
    }
```
It allows more flexibility for users to define their own commands.

**Comment by JanaJarecki** [comment]
got it. Thanks for the attention to detail
J


Am 25.02.2016 um 18:46 schrieb Randy Lai:

**Comment by JanaJarecki** [comment]
Hey, seems the near future has come. Neither the above command nor the below one do the job.
I'm on Windows 10, Sublimetext 3103.

If I try the (new/cool) r-box menu and "Source current file", the error is this:

> Fehler: '\U' ohne Hex-Ziffern in der Zeichenkette beginnend mit ""C:\U" genutzt
For the source codoe
> source("C:\Users\Jana Jarecki\....\filename.r")

Maybe it doesnt like the spaces?

**Comment by randy3k** [comment]
You are right. Will fix it soon.

**Comment by randy3k** [comment]
Fix in SendText+ 0.6.3.
